exotic/exotic.py

```diff
@@ -713,43 +713,78 @@ def deg_to_pix(exp_ra, exp_dec, ra_list, dec_list):
-                raise ValueError
-            if not (expdec - uncert <= declist[int(pixy)][int(pixx)] <= expdec + uncert):
-                log_info("\nWarning: The Y Pixel Coordinate entered does not match the target's Declination.", warn=True)
-                raise ValueError
-            return pixx, pixy
+            validate_pixel_coordinates(input_x_pixel, input_y_pixel, centroid_x, centroid_y, sigma_x, sigma_y)
+            return input_x_pixel, input_y_pixel
         except ValueError:
-            log_info(f"Your input pixel coordinates: [{pixx}, {pixy}]")
-            exotic_pixy, exotic_pixx = deg_to_pix(expra, expdec, ralist, declist)
-            log_info(f"EXOTIC's calculated pixel coordinates: [{exotic_pixx}, {exotic_pixy}]")
-
-            opt = user_input("Would you like to re-enter the pixel coordinates? (y/n): ", type_=str, values=['y', 'n'])
-
-            # User wants to change their coordinates
-            if opt == 'y':
-                searchopt = user_input(f"Here are the suggested pixel coordinates:"
-                                       f"  X Pixel: {exotic_pixx}"
-                                       f"  Y Pixel: {exotic_pixy}"
-                                       "\nWould you like to use these? (y/n): ",
-                                       type_=str, values=['y', 'n'])
-                # Use the coordinates found by code
-                if searchopt == 'y':
-                    return exotic_pixx, exotic_pixy
-                # User enters their own coordinates to be re-checked
-                else:
-                    pixx = user_input("Please re-enter the target star's X Pixel Coordinate: ", type_=int)
-                    pixy = user_input("Please re-enter the target star's Y Pixel Coordinate: ", type_=int)
+            new_x_pixel, new_y_pixel = prompt_user_for_coordinates(input_x_pixel, input_y_pixel,
+                                                                   calculated_x_pixel, calculated_y_pixel)
+            if new_x_pixel == input_x_pixel and new_y_pixel == input_y_pixel:
+                return input_x_pixel, input_y_pixel
             else:
-                # User does not want to change coordinates even though they don't match the expected ra and dec
-                return pixx, pixy
+                input_x_pixel, input_y_pixel = new_x_pixel, new_y_pixel
+
+
+def validate_pixel_coordinates(input_x_pixel, input_y_pixel, centroid_x, centroid_y, sigma_x, sigma_y):
+    """
+    Validating the provided pixel coordinates are within 5 PSF of the expected coordinates.
+    """
+    x_min = centroid_x - (sigma_x * 5)
+    x_max = centroid_x + (sigma_x * 5)
+    y_min = centroid_y - (sigma_y * 5)
+    y_max = centroid_y + (sigma_y * 5)
+
+    if not (x_min <= input_x_pixel <= x_max):
+        log_info("\nWarning: The X Pixel Coordinate entered does not match the target's Right Ascension.", warn=True)
+        raise ValueError
+    if not (y_min <= input_y_pixel <= y_max):
```

Comment:
Is it OK if they're all the same value?

Reply:
Technically, yes. If somehow `x_min`, `input_x_pixel`, and `x_max` all contain values of 5, indicating that there is no "spread" in the point-source and that there is only one pixel position that is correct, then I'm okay with it not executing. It's unlikely, but I'm still okay with it occurring.